py-polars/tests/unit/operations/namespaces/string/test_string.py

```diff
@@ -1061,6 +1066,88 @@ def test_replace_many(
     )
 
 
+@pytest.mark.parametrize(
+    ("mapping", "case_insensitive", "expected"),
+    [
+        ({}, False, "Tell me what you want"),
+        ({"me": "them"}, False, "Tell them what you want"),
+        ({"who": "them"}, False, "Tell me what you want"),
+        ({"me": "it", "you": "it"}, False, "Tell it what it want"),
+        ({"Me": "it", "you": "it"}, False, "Tell me what it want"),
+        ({"me": "you", "you": "me"}, False, "Tell you what me want"),
+        ({}, True, "Tell me what you want"),
+        ({"Me": "it", "you": "it"}, True, "Tell it what it want"),
+        ({"me": "you", "YOU": "me"}, True, "Tell you what me want"),
+    ],
+)
+def test_replace_many_mapping(
+    mapping: dict[str, str],
+    case_insensitive: bool,
+    expected: str,
+) -> None:
+    df = pl.DataFrame({"text": ["Tell me what you want"]})
+    # series
+    assert (
+        expected
+        == df["text"]
+        .str.replace_many(mapping, ascii_case_insensitive=case_insensitive)
+        .item()
+    )
+    # expr
+    assert (
+        expected
+        == df.select(
+            pl.col("text").str.replace_many(
+                mapping,
+                ascii_case_insensitive=case_insensitive,
+            )
+        ).item()
+    )
+
+
+def test_replace_many_invalid_inputs() -> None:
```

Comment:
This could alternatively be moved to `py-polars/tests/unit/test_errors.py` if so desired.